libflux/flux-core/src/semantic/convert.rs

```diff
@@ -450,16 +501,54 @@ impl<'a> Converter<'a> {
                 (id.name.clone(), Some(id))
             }
         };
-        let path = self.convert_string_literal(&imp.path);
+        let (registry, path) = self.convert_import_path(&imp.path);
 
         ImportDeclaration {
             loc: imp.base.location.clone(),
             alias,
+            registry,
             path,
             import_symbol,
         }
     }
 
+    fn convert_import_path(&mut self, path: &ast::StringLit) -> (Option<String>, String) {
+        let path_str = path.value.as_str();
+        // Empty components or components completely composed of dots are just not allowed.
+        for p in path_str.split('/') {
+            if p.is_empty() || p == "." || p == ".." {
+                self.errors.push(located(
+                    path.base.location.clone(),
+                    ErrorKind::InvalidImportPath(path_str.to_string()),
```

Comment:
This could explain why the path is invalid more precisely, either by pointing precisely to the invalid segment, or explaining why a segment is invalid.

Reply:
I'll give this a try.

Reply:
Ok I changed the error message to be specific about which path segment. I couldn't really get it to point to the exact location, but it should have the location of the import itself and the path segment that was invalid.